migrate/migrate.go

```diff
@@ -99,14 +100,14 @@ func NewMigratorEx(ctx context.Context, conn *pgx.Conn, versionTable string, opt
 }
 
 type MigratorFS interface {
-	ReadDir(dirname string) ([]os.FileInfo, error)
+	ReadDir(dirname string) ([]fs.FileInfo, error)
```

Comment:
This is the key change, if you don't like anything from this patch I can drop all but this change. This allows me to write the FS implementation on our side.